### PR TITLE
Delete the obsolete CoreFxNetCloudService

### LINKED ISSUE #52693: Improve kestrel version of networking test echo server (previously CoreFxNetCloudService)
As part of https://github.com/dotnet/runtime/issues/42852 we are bringing kestrel version of the [CoreFxNetCloudService](https://github.com/dotnet/runtime/tree/f309b53d4e435dbe4a7945676d167ef910959403/src/libraries/Common/tests/System/Net/Prerequisites/Servers/CoreFxNetCloudService) echo test server. The proposed implementation is missing:
- Any deployment logic to be able to host it in Azure- `ClientCertificate` capturing in 
- Any test certificates, which the unit tests may expect
- `NetCoreServer\Helpers\RequestInformation.cs` and necessary setup of kestrel to receive it. See [this conversation](https://github.com/dotnet/runtime/pull/52642#discussion_r631116284)

### PR DESCRIPTION
**Pull request by pedrobsaila**
Fixes #52693

### CODE REVIEW

File: src/libraries/Common/tests/System/Net/Prerequisites/LocalEchoServer.props [hunk 1]
```diff
@@ -5,10 +5,6 @@
     <TestArchiveTestsDir>$(TestArchiveTestsRoot)$(OSPlatformConfig)/</TestArchiveTestsDir>
 
     <!-- handle different path to middleware in Helix -->
-    <_TestEchoMiddleware Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(OS)' == 'Windows_NT'">%HELIX_CORRELATION_PAYLOAD%/xharness/TestEchoMiddleware</_TestEchoMiddleware>
```
[the remark this answers is not in the document]
Comment (pedrobsaila) [reply src/libraries/Common/tests/System/Net/Prerequisites/LocalEchoServer.props:8]: TestEcho isn't related to NetCoreServer ? I mean if we choose to keep it, don't we need to keep NetCoreServer also ?

**Comment by pavelsavara** [comment]
This is the old one https://github.com/dotnet/runtime/tree/f309b53d4e435dbe4a7945676d167ef910959403/src/libraries/Common/tests/System/Net/Prerequisites/Servers/CoreFxNetCloudService

The one this PR deleted is in use.

I also wonder if that old one could be really deleted, because that's what's deployed to Azure for coreCLR outerloop tests. This PR doesn't address deployment and other concerns raised in the linked issues.

**Comment by pedrobsaila** [comment]
@pavelsavara my bad sorry, I mistook the class.
I didn't address the other concerns because of the follwoing comment in the issue :

Triage: We should delete the old code as it is unused.
We can ignore the rest, until it is needed.

**Comment by pavelsavara** [comment]
@karelz is the `CoreFxNetCloudService` really unused at the moment ? Or is that what lives in Azure right now ?

@pedrobsaila No worries, I will close this one and you could try the other one, based on Karel's answer.

**Pull request closed without merging**